tests/ut/ir/statements/test_scope_stmt_hierarchy.py

```
@@ -218,6 +218,37 @@ def test_printer_incore_scope_unchanged():
+def test_structural_equal_incore_with_split():
+    """structural_equal compares split on InCore ScopeStmt."""
+    s1 = ir.ScopeStmt(ir.ScopeKind.InCore, _empty_body(), _span(), split=ir.SplitMode.UP_DOWN)
+    s2 = ir.ScopeStmt(ir.ScopeKind.InCore, _empty_body(), _span(), split=ir.SplitMode.UP_DOWN)
+    ir.assert_structural_equal(s1, s2)
+
+
+def test_structural_equal_incore_different_split():
+    """structural_equal detects different split modes."""
+    s1 = ir.ScopeStmt(ir.ScopeKind.InCore, _empty_body(), _span(), split=ir.SplitMode.UP_DOWN)
+    s2 = ir.ScopeStmt(ir.ScopeKind.InCore, _empty_body(), _span(), split=ir.SplitMode.LEFT_RIGHT)
+    with pytest.raises(Exception):
```

Ruff (0.15.9)

[warning] Do not assert blind exception: `Exception`

(B017)